### PR TITLE
[Improve] Flink version check improvement

### LINKED ISSUE #2563: [Bug] Class conflict when stopping application and doing savepoint
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

1.8

### Scala Version

2.12.x

### StreamPark Version

2.1.0

### Flink Version

1.16.0

### deploy mode

None

### What happened

Class conflict when stopping application and doing savepoint

### Error Exception

```log
java.util.concurrent.CompletionException: java.lang.reflect.InvocationTargetException
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1606)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.streampark.flink.client.FlinkClient$.$anonfun$proxy$1(FlinkClient.scala:68)
	at org.apache.streampark.flink.proxy.FlinkShimsProxy$.$anonfun$proxy$1(FlinkShimsProxy.scala:79)
	at org.apache.streampark.common.util.ClassLoaderUtils$.runAsClassLoader(ClassLoaderUtils.scala:40)
	at org.apache.streampark.flink.proxy.FlinkShimsProxy$.proxy(FlinkShimsProxy.scala:79)
	at org.apache.streampark.flink.client.FlinkClient$.proxy(FlinkClient.scala:63)
	at org.apache.streampark.flink.client.FlinkClient$.cancel(FlinkClient.scala:47)
	at org.apache.streampark.flink.client.FlinkClient.cancel(FlinkClient.scala)
	at org.apache.streampark.console.core.service.impl.ApplicationServiceImpl.lambda$cancel$6(ApplicationServiceImpl.java:1251)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604)
	... 3 more
Caused by: java.lang.NoSuchMethodError: org.apache.flink.client.program.ClusterClient.cancelWithSavepoint(Lorg/apache/flink/api/common/JobID;Ljava/lang/String;)Ljava/util/concurrent/CompletableFuture;
	at org.apache.streampark.flink.core.FlinkClientTrait.cancelWithSavepoint(FlinkClientTrait.scala:32)
	at org.apache.streampark.flink.client.trait.FlinkClientTrait.cancelJob(FlinkClientTrait.scala:486)
	at org.apache.streampark.flink.client.trait.FlinkClientTrait.cancelJob$(FlinkClientTrait.scala:474)
	at org.apache.streampark.flink.client.impl.YarnPerJobClient$.org$apache$streampark$flink$client$trait$YarnClientTrait$$super$cancelJob(YarnPerJobClient.scala:41)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.$anonfun$doCancel$1(YarnClientTrait.scala:72)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.$anonfun$executeClientAction$1(YarnClientTrait.scala:56)
	at scala.util.Try$.apply(Try.scala:209)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.executeClientAction(YarnClientTrait.scala:56)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.doCancel(YarnClientTrait.scala:71)
	at org.apache.streampark.flink.client.trait.YarnClientTrait.doCancel$(YarnClientTrait.scala:70)
	at org.apache.streampark.flink.client.impl.YarnPerJobClient$.doCancel(YarnPerJobClient.scala:123)
	at org.apache.streampark.flink.client.trait.FlinkClientTrait.cancel(FlinkClientTrait.scala:179)
	at org.apache.streampark.flink.client.trait.FlinkClientTrait.cancel$(FlinkClientTrait.scala:163)
	at org.apache.streampark.flink.client.impl.YarnPerJobClient$.cancel(YarnPerJobClient.scala:41)
	at org.apache.streampark.flink.client.FlinkClientHandler$.cancel(FlinkClientHandler.scala:47)
	at org.apache.streampark.flink.client.FlinkClientHandler.cancel(FlinkClientHandler.scala)
	... 16 more
```

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wolfboys**
[Improve] Flink version check improvement

## What changes were proposed in this pull request

Issue Number: close #2563

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): no

### CODE REVIEW

**Comment by wolfboys** [comment]
cc @lvshaokang @zhoulii please review and test. thx.

**Comment by zhoulii** [comment]
@wolfboys Nice work. Tested with flink 1.12.5/1.15.3/1.16.0/1.17.0 remote mode, it works well.

File: streampark-common/src/main/scala/org/apache/streampark/common/conf/FlinkVersion.scala [hunk 1]
```diff
@@ -109,6 +109,18 @@ class FlinkVersion(val flinkHome: String) extends java.io.Serializable with Logg
     distJar.head
   }
 
+  def checkVersion(throwable: Boolean = true): Boolean = {
```
Comment (lvshaokang) [inline streampark-common/src/main/scala/org/apache/streampark/common/conf/FlinkVersion.scala]:
```suggestion
  def checkVersion(throwException: Boolean = true): Boolean = {
```

Comment (wolfboys) [reply streampark-common/src/main/scala/org/apache/streampark/common/conf/FlinkVersion.scala]: > 

updated

**Review by lvshaokang — APPROVED** [review]
LGTM

**Pull request merged**